src/parallel/merge-engine.ts

```
@@ -0,0 +1,657 @@
+    throw new Error(`Invalid git ref for ${context}: ref is empty`);
+  }
+  // Cannot contain spaces
+  if (ref.includes(' ')) {
+    throw new Error(`Invalid git ref for ${context}: contains spaces`);
+  }
+  // Cannot contain double dots
+  if (ref.includes('..')) {
+    throw new Error(`Invalid git ref for ${context}: contains '..'`);
+  }
+  // Cannot contain control characters
+  if (/[\x00-\x1f\x7f]/.test(ref)) {
```

Biome (2.3.13)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)